api/v1/certificatemanagement_types.go

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2020 Tigera, Inc. All rights reserved.
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, softwa
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CertificateManagementStatus defines the observed state of CertificateManagement
+type CertificateManagementStatus struct {
+	// State provides user-readable status.
+	State string `json:"state,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster
+
+// CertificateManagement is the Schema for the certificate management API
+type CertificateManagement struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   CertificateManagementSpec   `json:"spec,omitempty"`
+	Status CertificateManagementStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// CertificateManagementList contains a list of CertificateManagement
+type CertificateManagementList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []CertificateManagement `json:"items"`
+}
+
+func init() {
+	SchemeBuilder.Register(&CertificateManagement{}, &CertificateManagementList{})
+}
+
+// Apply this to your cluster only if an automatic certificate signing and approving process is in place for the cluster.
+// Rather than creating self-signed certificates, pods created by this operator will issue a CSR. These pods will not
+// come up until their CSRs are approved.
+type CertificateManagementSpec struct {
+	// Root CA of the certificate authority that signs the certificate requests.
+	RootCA string `json:"rootCA"`
```

Comment:
Is this the root cert (possible cert chain) from the CA? It feels a little odd to me that this needs to be provided and doesn't come with the signed certificate. But maybe this is so the signed cert can be verified it came from the right source?
I'm not suggesting change but mentioning it seems odd to me. Though I certainly am not super knowledgeable about cert/signing.